client/server/middleware/build-target.js

```diff
@@ -5,11 +5,16 @@ import { matchesUA } from 'browserslist-useragent';
  *
  * @param {string} userAgentString The user agent string.
  * @param {string} environment The `browserslist` environment.
- *
  * @returns {boolean} Whether the user agent is included in the browser list.
  */
 function isUAInBrowserslist( userAgentString, environment = 'defaults' ) {
-	return matchesUA( userAgentString, {
+	// If the user agent string includes Electron, it probably comes from the
+	// desktop app. Unfortunately, browserslist then parses the UA family to be
+	// "WordPress". We need to make sure we test against the Electron version
+	// instead, or the desktop app won't be considered a supported browser.
+	const electronUA = userAgentString.match( /(Electron\/[0-9.]*\S)/ );
```

Comment:
Given the user agent string `Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4_0) AppleWebKit/537.36 (KHTML, like Gecko) WordPressDesktop/7.1.0 Chrome/89.0.4389.128 Electron/12.0.13 Safari/537.36`, 

the `resolveUserAgent` function, which is part of `matchesUA`, will return `{ family: 'WordPress', version: '0.0.0' }`. Since `browserslist` doesn't know about "wordpress", we can't just pass in the UA string directly. Instead, we have to parse out the Electron portion of this string. Everything else is basically irrelevant to the perspective of browserslist (which will convert it to a Chrome version under the hood)

Reply:
I have another solution that doesn't require a `browserslist-useragent` patch. The UA string in your comment already contains a `Chrome/89.0.4389.128` string, so why not let `browserslist-useragent` (and the `useragent` package it uses) parse that?

The `useragent` does a `for` loop through an array of ~260 regexps and returns the first match. And there are two regexps that trigger a result we don't want, and both of them are before the desired Chrome one in that array:
- one that searches for `WordPress` in the UA and detects such a match as a bot. There is apparently some crawler bot, or there was one 7 years ago when this regexp was added, that has `WordPress` in its UA string. Maybe it's even an Automattic bot 🙂 
- one that searches for `Electron` and detects Electron.

We can strip these two undesired tokens by sanitizing the UA string:
```js
const uaFixed = ua.replace( / (WordPressDesktop|Electron)\/[.\d]+/g, '' );
```
and only then passing it to `matchesUA`.

My local testing shows that parsing the sanitized UA indeed yields exactly the result we want:
```
{ family: 'Chrome', major: '89', minor: '0', patch: '4389' }
```

Reply:
This approach sounds reasonable to me. 😄 

On a somewhat tangential note - I'm wondering whether this might affect Tracks reporting for the Desktop app. Specifically, in the Tracks dashboard, the desktop app is differentiated via the browser family "WordPress", and I'm guessing it comes from the user agent parsing as mentioned above. 

At the moment I don't know how Tracks events work in Calypso, exactly - but my guess is _somewhere_ the user agent is resolved to `WordPress` before any events are fired (I'm also not sure if Tracks events fire via the server or the browser).

Perhaps this isn't a concern at all, but it just occurred to me that it's probably something we want to double-check.